feat: [002-test code, transaction]

karpei-taemukan opened the pull request:
### 변경사항

- 입금과 출금 에 대한 컨트롤러와 서비스 작성 (TransactionController, TransactionService)

- 컨트롤러와 서비스 에 대한 테스트 코드 작성

밑에 작성한 부분들만 새로 작성했습니다

   - 컨트롤러 -> SignUpControllerTest, SignInControllerTest, TransactionControllerTest

   - 서비스 -> SignUpServiceTest, SignInServiceTest, TransactionServiceTest, VerifyServiceTest

karpei-taemukan committed:
feat: [002-test code, transaction]

spring-kotlin-lover commented on src/main/java/com/zerobase/controller/TransactionController.java:
```diff
@@ -0,0 +1,51 @@
+  @PostMapping("/deposit")
+  private ResponseEntity<AccountDto> depositAccount(
+      @RequestBody @Valid TransactionForm form,
+      @RequestHeader(name = "Authorization") String token
```
Authorization Header 기반으로 token 정보는 파싱하고, 파싱된 token 정보를 통해 email 을 가져오는 로직이 controller  쪽에서 계속 중복될 것으로 보이네요.

어떻게 하면 중복 코드를 없애볼 수 있을까요?

spring-kotlin-lover commented on src/main/java/com/zerobase/service/AccountService.java:
```diff
@@ -1,7 +1,35 @@
+  private final AccountUserRepository accountUserRepository;
+  private final AccountRepository accountRepository;
+
+  public void searchAllAccount(String email, SearchForm form) {
```
search 라고 되어 있는데요. return 값이 void type 인 이유는 무엇일까요? 주석에 따르면 모든 계좌를 보여준다고 하는데, 어떤 의도로 설계된 메서드인지 이해하기가 어렵네요

spring-kotlin-lover commented on src/main/java/com/zerobase/domain/TransactionForm.java:
```diff
@@ -0,0 +1,16 @@
+public class TransactionForm {
+
+  private String accountName;
+  private Long balance;
```
validation 관련된 어노테이션이 없는데 정의해볼 수 있을까요? 만일 balance 가 음수로 들어오면 어떻게 될까요?

karpei-taemukan committed:
feat: [003 - Account Service, testCode]

karpei-taemukan committed:
feat: [004 - add ArgumentResolver]

1. ArgumentResolver 추가에 따라 configuration 추가(WebConfig), dto(TokenDto), TokenArgumentResolver 추가

2. ArgumentResolver 가 적용된 Controller
-> TransactionController, AccountController

3. 바뀐 Controller 에 따라 테스트 코드도 변경
